Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,155 @@
+            onProbChanged = new UnityEvent();
+            onGameActiveToggle = new UnityEvent();
+            onRestartClick = new UnityEvent();
+            onSizeChanged.AddListener(() => UpdateParam(ValuesEnum.Size));
+            onSpeedChanged.AddListener(() => UpdateParam(ValuesEnum.Speed));
+            onProbChanged.AddListener(() => UpdateParam(ValuesEnum.Prob));
+            onGameActiveToggle.AddListener(() =>
+            {
+                gameActive = !gameActive;
+                if (gameActive) StartCoroutine(GameFunction());
+                renderToggle(ValuesEnum.Prob);
+                renderToggle(ValuesEnum.Size);
+            });
+            onRestartClick.AddListener(() =>
+            {
+                generation = 1;
+                grid.OnRestart(getSize(),getProb());
+            });
+        }
+
+         int getMaximalSize()
+        {
+            return sizelarge;
+        }
+
+        void UpdateParam(ValuesEnum valueToChange)
+        {
+            if (valueToChange == ValuesEnum.Size)
+            {
+                int oldSize = options.GetCurrentValue(ValuesEnum.Size);
+                options.GetNextValue(ValuesEnum.Size);
+                grid.UpdateGridBySize(getSize(),oldSize);
+            }
+            else
+            {
+                options.GetNextValue(valueToChange);
+            }
+
+            if (valueToChange != ValuesEnum.Speed)
+            {
+                generation = 1;
+                grid.OnRestart(getSize(),getProb());
+            }
+
+            renderToggle(valueToChange);
+        }
+
+
+        IEnumerator GameFunction()
+        {
+            while (gameActive)
+            {
+                yield return new WaitForSeconds(getSpeed() / 1000);
+                aliveCounter = grid.UpdateGridOnGenerationChange(getSize());
+                generation += 1;
+            }
+        }
+
+        private int getSpeed()
+        {
+            return options.GetCurrentValue(ValuesEnum.Speed);
+        }
+
+        private int getSize()
+        {
+            return options.GetCurrentValue(ValuesEnum.Size);
+        }
+
+        private int getProb()
+        {
+            return options.GetCurrentValue(ValuesEnum.Prob);
+        }
+
+        private void renderToggle(ValuesEnum value)
+        {
+            switch (value)
+            {
+                case ValuesEnum.Prob:
+                {
+                    probToggle.Render(ValuesEnum.Prob, getProb(), onProbChanged,gameActive);
+                    break;
+                }
+                case ValuesEnum.Size:
+                {
+                    sizeToggle.Render(ValuesEnum.Size, getSize(), onSizeChanged,gameActive);
+                    break;
+                }
+                default:
+                {
+                    speedToggle.Render(ValuesEnum.Speed, getSpeed(), onSpeedChanged,false);
+                    break;
+                }
+            }
+        }
+
+        void Update()
+        {
```

Comment:
it will be even better if instead of doing it on every Update(on every frame)
just do it when one of the values are changed
check if you can do it that way